File: web/src/lib/mockData.ts
/* COMMON §9 — mock data ported to TS, drives all pages. */

export type DeviceKind = 'mac' | 'ios' | 'ipad' | 'android' | 'win' | 'linux' | 'web'

export interface MeshDevice {
  id: string
  name: string
  who: string
  kind: DeviceKind
  dist: number
  angle: number
  color: string
  initials: string
  os: string
  rtt: number
  online?: boolean
  paired?: boolean
}

export const MESHDROP_DEVICES: MeshDevice[] = [
  { id: 'lily',   name: "Lily's MacBook",   who: '李莉',   kind: 'mac',     dist: 0.55, angle: 35,  color: '#FFB4A1', initials: 'LL', os: 'macOS 14',  rtt: 18, online: true,  paired: true  },
  { id: 'kun',    name: 'Kun · Pixel 8',    who: '坤',     kind: 'android', dist: 0.78, angle: 110, color: '#B7E5C8', initials: 'K',  os: 'Android 15', rtt: 32, online: true,  paired: true  },
  { id: 'jiawei', name: 'Jiawei · iPad',    who: '嘉伟',   kind: 'ipad',    dist: 0.40, angle: 200, color: '#C7B8FF', initials: 'JW', os: 'iPadOS 18', rtt: 14, online: true,  paired: true  },
  { id: 'mengxi', name: 'Meng Xi · iPhone', who: '孟茜',   kind: 'ios',     dist: 0.62, angle: 265, color: '#FFD970', initials: 'MX', os: 'iOS 18',    rtt: 26, online: true,  paired: true  },
  { id: 'dev01',  name: 'DEV-01 · Win 11',  who: '工位机', kind: 'win',     dist: 0.88, angle: 320, color: '#9AD0FF', initials: 'D1', os: 'Win 11',    rtt: 41, online: false, paired: false },
]

export interface HistoryEntry {
  id: string
  dir: 'incoming' | 'outgoing'
  peer: string
  time: string
  kind: 'image' | 'file' | 'text'
  status: 'done' | 'transferring' | 'queued' | 'failed'
  name?: string
  size?: string
  ext?: string
  count?: number
  content?: string
  progress?: number
}

export const MESHDROP_HISTORY: HistoryEntry[] = [
  { id: 'h6', dir: 'incoming', peer: '孟茜',   time: '14:18', kind: 'image', count: 2, status: 'done' },
  { id: 'h5', dir: 'outgoing', peer: '孟茜',   time: '14:10', kind: 'file',  name: '设计稿_v3_final.fig', size: '14.2 MB', ext: 'fig', status: 'done' },
  { id: 'h4', dir: 'outgoing', peer: '李莉',   time: '14:09', kind: 'text',  content: '改完了，整理一下发你 👇', status: 'done' },
  { id: 'h3', dir: 'outgoing', peer: '嘉伟',   time: '14:08', kind: 'file',  name: 'iOS-mocks-final.zip', size: '48.6 MB', ext: 'zip',  progress: 67, status: 'transferring' },
  { id: 'h2', dir: 'incoming', peer: '坤',     time: '13:58', kind: 'file',  name: 'IMG_4821~38.heic',    size: '128 MB',  ext: 'heic', progress: 12, status: 'transferring' },
  { id: 'h1', dir: 'outgoing', peer: '李莉',   time: '13:42', kind: 'file',  name: 'demo-video.mp4',      size: '512 MB',  ext: 'mp4',  status: 'queued' },
]

export interface PendingPairing {
  id: string
  peer: string
  deviceName: string
  fingerprint: string
  receivedAt: string
}

export const MESHDROP_PENDING_PAIRING: PendingPairing = {
  id: 'pp-1',
  peer: '李莉',
  deviceName: "Lily's MacBook",
  fingerprint: 'ZX8K · L72M · 9FQ3 · 7HD2 · M1P6 · QA8N · KZ9R · X3WF',
  receivedAt: '8s ago',
}

export interface PendingOffer {
  id: string
  peer: string
  deviceName: string
  fileName: string
  fileSize: string
  ext?: string
  pages?: number
  note?: string
  receivedAt: string
}

export const MESHDROP_PENDING_OFFER: PendingOffer = {
  id: 'po-1',
  peer: '嘉伟',
  deviceName: 'Jiawei · iPad',
  fileName: '规划文档_v0.3.pages',
  fileSize: '3.4 MB',
  ext: 'pages',
  pages: 12,
  note: '改完了帮我看下第二章，特别是 §2.3 那段',
  receivedAt: 'just now',
}

export interface ClipboardItem {
  id: string
  who: string
  kind: 'link' | 'text' | 'code'
  body: string
  ago: string
  lang?: string
}

export const MESHDROP_CLIPBOARD: ClipboardItem[] = [
  { id: 'cb1', who: '嘉伟', kind: 'link', body: 'https://internal.acme.io/specs/auth-v3', ago: '8s' },
  { id: 'cb2', who: '孟茜', kind: 'text', body: '"1. 新流程要支持端到端\n2. 雷达扫描频率调到 2s\n3. iPad 端做横屏适配"', ago: '12m' },
  { id: 'cb3', who: '李莉', kind: 'code', lang: 'sh', body: 'docker run --rm -v $PWD:/app meshdrop/build:latest', ago: '34m' },
  { id: 'cb4', who: '坤',   kind: 'text', body: '"会议室 B 已订到 16:00–17:30"', ago: '1h' },
  { id: 'cb5', who: '我',   kind: 'link', body: 'figma://file/Q8xK2/MeshDrop?node-id=42:108', ago: '2h' },
]

export type TransferState = 'sending' | 'receiving' | 'done' | 'failed' | 'queued'

export interface TransferRow {
  id: string
  name: string
  size: string
  ext: string
  from: string
  to: string
  progress: number
  state: TransferState
  eta?: string
  speed?: string
}

export const MESHDROP_TRANSFERS: TransferRow[] = [
  { id: 't1', name: '设计稿_v3_final.fig',  size: '14.2 MB',         ext: 'fig',  from: '我',  to: '孟茜', progress: 100, state: 'done',       eta: '00:08' },
  { id: 't2', name: 'iOS-mocks-final.zip',  size: '48.6 MB',         ext: 'zip',  from: '我',  to: '孟茜', progress: 67,  state: 'sending',    speed: '8.4 MB/s',  eta: '00:02' },
  { id: 't3', name: 'spec_PRD_2026Q1.pdf',  size: '2.1 MB',          ext: 'pdf',  from: '我',  to: '嘉伟', progress: 34,  state: 'sending',    speed: '3.1 MB/s',  eta: '00:01' },
  { id: 't4', name: 'IMG_4821~IMG_4838.heic', size: '128 MB · 18 张', ext: 'heic', from: '坤',  to: '我',  progress: 12,  state: 'receiving',  speed: '11.7 MB/s', eta: '00:09' },
  { id: 't5', name: 'release-notes.md',     size: '4.8 KB',          ext: 'md',   from: '我',  to: 'DEV-01', progress: 100, state: 'done',     eta: '00:01' },
  { id: 't6', name: 'demo-video.mp4',       size: '512 MB',          ext: 'mp4',  from: '我',  to: '李莉', progress: 0,   state: 'queued' },
]

export const MESHDROP_UPLOAD_BARS = [3, 5, 8, 7, 9, 6, 11, 12, 14, 11, 10, 11, 12, 11]
export const MESHDROP_DOWNLOAD_BARS = [8, 9, 7, 6, 5, 7, 10, 12, 11, 12, 11, 12, 11, 12]
export const MESHDROP_SESSION_BARS = [2, 3, 5, 4, 6, 8, 7, 9, 10, 12, 11, 12, 11, 12, 14]

export const MESHDROP_ME = {
  name: '访客 · Visitor',
  fingerprint: 'ZX8K · L72M · 9FQ3 · 7HD2',
  ip: '192.168.1.207',
  hostIp: '192.168.1.42',
  os: 'Chromium 121 · macOS',
  visibility: '访客可见',
  pairingCode: 'XJ9-LM4',
}

export interface HistoryDay {
  label: string
  items: HistoryEntry[]
}

export const MESHDROP_HISTORY_BY_DAY: HistoryDay[] = [
  {
    label: 'TODAY · 今天 · 6 件',
    items: MESHDROP_HISTORY,
  },
  {
    label: 'YESTERDAY · 昨天 · 4 件',
    items: [
      { id: 'y1', dir: 'incoming', peer: '李莉', time: '昨 22:14', kind: 'file',  name: 'meeting-notes.md', size: '8.2 KB',  ext: 'md',  status: 'done' },
      { id: 'y2', dir: 'outgoing', peer: '坤',   time: '昨 19:31', kind: 'image', count: 3, status: 'done' },
      { id: 'y3', dir: 'incoming', peer: '嘉伟', time: '昨 18:02', kind: 'text',  content: '后端接口我接好了，明早一起跑下', status: 'done' },
      { id: 'y4', dir: 'outgoing', peer: '孟茜', time: '昨 09:48', kind: 'file',  name: 'logo-final.svg', size: '46 KB',  ext: 'svg', status: 'done' },
    ],
  },
  {
    label: 'EARLIER · 早些时候 · 3 件',
    items: [
      { id: 'e1', dir: 'incoming', peer: '孟茜', time: '周三 16:20', kind: 'file', name: '团队合影.heic', size: '4.1 MB',  ext: 'heic', status: 'done' },
      { id: 'e2', dir: 'outgoing', peer: 'DEV-01', time: '周二 14:08', kind: 'file', name: 'build.exe', size: '92 MB',  ext: 'exe', status: 'failed' },
      { id: 'e3', dir: 'outgoing', peer: '李莉',  time: '周二 11:30', kind: 'text', content: '中午一起？三楼那家', status: 'done' },
    ],
  },
]

File: windows/MeshDrop/Mock/MockData.cs
using System.Collections.Generic;
using System.Collections.ObjectModel;

namespace MeshDrop.Mock;

public enum MockKind { Mac, Win, Ipad, Ios, Android, Linux }

public enum MockTransferState
{
    Queued,
    Sending,
    Receiving,
    Done,
    Failed,
}

public enum MockHistoryKind { Text, File, Image }
public enum MockClipboardKind { Text, Link, Code }

public sealed record MockDevice(
    string Id,
    string Name,
    string Who,
    MockKind Kind,
    double Dist,           // 0..1
    int Angle,             // 0..359
    string ColorHex,
    string Initials,
    string Os,
    int Rtt,
    string Fingerprint = "ZX8K · L72M · 9FQ3 · 7HD2 · M1P6 · QA8N · KZ9R · X3WF",
    string Ip = "192.168.1.31",
    string Bw = "940 Mbps",
    bool E2EVerified = true);

public sealed record MockHistory(
    string Id,
    string Dir,               // "incoming" | "outgoing"
    string Peer,
    string Time,
    MockHistoryKind Kind,
    string? Name = null,
    string? Size = null,
    string? Ext = null,
    string? Content = null,
    int? Count = null,
    int? Progress = null,
    string Status = "done")
{
    public string DirArrow => Dir == "outgoing" ? "↑" : "↓";

    public string KindLabel => Kind switch
    {
        MockHistoryKind.Text => "TEXT",
        MockHistoryKind.File => Ext is null ? "FILE" : $"FILE · {Ext.ToUpperInvariant()}",
        MockHistoryKind.Image => Count is int n ? $"IMAGE · {n} 张" : "IMAGE",
        _ => "TEXT",
    };

    public string Summary => Kind switch
    {
        MockHistoryKind.Text => Content ?? "",
        MockHistoryKind.File => $"{Name ?? "(unknown)"}  ·  {Size ?? ""}",
        MockHistoryKind.Image => Count is int n ? $"{n} 张图片" : "图片",
        _ => "",
    };

    public string StatusLabel => Status switch
    {
        "done" => "DONE",
        "transferring" => Progress is int p ? $"{p}%" : "活跃",
        "queued" => "QUEUED",
        "failed" => "FAILED",
        _ => Status.ToUpperInvariant(),
    };

    public ChipToneName StatusTone => Status switch
    {
        "done" => ChipToneName.Lime,
        "transferring" => Dir == "outgoing" ? ChipToneName.Flame : ChipToneName.Sky,
        "failed" => ChipToneName.Error,
        _ => ChipToneName.Outline,
    };
}

/// <summary>独立 chip tone 名（避免直接引 Views.Controls 命名空间，免 mock 依赖 UI）。</summary>
public enum ChipToneName { Mute, Lime, Ink, Outline, Flame, Sky, Error }

public sealed record MockPendingPairing(
    string Id,
    string Peer,
    string DeviceName,
    string Fingerprint,
    string ReceivedAt);

public sealed record MockPendingOffer(
    string Id,
    string Peer,
    string DeviceName,
    string FileName,
    string FileSize,
    string Note,
    string ReceivedAt);

public sealed record MockClipboard(
    string Id,
    string Who,
    MockClipboardKind Kind,
    string Body,
    string Ago,
    string? Lang = null)
{
    public string KindLabel => Kind switch
    {
        MockClipboardKind.Text => "TEXT",
        MockClipboardKind.Link => "LINK",
        MockClipboardKind.Code => Lang is null ? "CODE" : $"CODE · {Lang.ToUpperInvariant()}",
        _ => "TEXT",
    };
}

public sealed record MockTransfer(
    string Name,
    string Size,
    string Ext,
    string From,
    string To,
    int Progress,
    MockTransferState State,
    string? Speed = null,
    string Eta = "")
{
    public string FromToText => $"{From} → {To}";
}

public sealed record MockMessage(
    string Id,
    string Side,              // "in" | "out"
    string Kind,              // "text" | "file" | "image"
    string? Text = null,
    string? Time = null,
    bool Delivered = true,
    string? FileName = null,
    string? FileSize = null,
    string? FileExt = null);

public sealed record MockMe(
    string Name = "DEV-01 · Win 11",
    string Fingerprint = "ZX8K · L72M · 9FQ3 · 7HD2",
    string Ip = "192.168.1.42",
    string Os = "Win 11",
    string Visibility = "可见",
    string Lan = "ACME-LAN");

public sealed record MockTrust(
    string Who,
    string DeviceName,
    string Os,
    string Fingerprint,
    string PairedOn,
    string LastSeen,
    bool RememberAllowed);

/// <summary>
/// COMMON §9 设计语言数据样本 —— 仅供 XAML designer preview 和单元测试用。
/// runtime 一律走 ShareEngine + EngineProjection，不要在 release path 引用本类。
/// </summary>
#if DEBUG
public static class MockData
{
    public static readonly MockMe Me = new();

    public static readonly IReadOnlyList<MockDevice> Devices = new[]
    {
        new MockDevice("lily",   "Lily's MacBook",   "李莉",   MockKind.Mac,     0.55, 35,  "#FFB4A1", "LL", "macOS 15.3", 18,
            Ip:"192.168.1.31", Bw:"940 Mbps (LAN)", Fingerprint:"ZX8K · L72M · 9FQ3 · 7HD2 · M1P6 · QA8N · KZ9R · X3WF"),
        new MockDevice("kun",    "Kun · Pixel 8",    "坤",     MockKind.Android, 0.78, 110, "#B7E5C8", "K",  "Pixel · Android 15", 32,
            Ip:"192.168.1.37", Bw:"540 Mbps", Fingerprint:"V2KN · 4HD8 · QX91 · LM3F · NP7B · K8RZ · F4WX · 2GLM"),
        new MockDevice("jiawei", "Jiawei · iPad",    "嘉伟",   MockKind.Ipad,    0.40, 200, "#C7B8FF", "JW", "iPadOS 18.2", 14,
            Ip:"192.168.1.44", Bw:"866 Mbps", Fingerprint:"BFM8 · 7KQ2 · X3LN · 9PRD · ZM4F · K8H1 · NQ6W · L2VR"),
        new MockDevice("mengxi", "Meng Xi · iPhone", "孟茜",   MockKind.Ios,     0.62, 265, "#FFD970", "MX", "iOS 18.2", 26,
            Ip:"192.168.1.49", Bw:"700 Mbps", Fingerprint:"H7LZ · 2K9F · QM48 · X1DR · NB5P · FW3K · L8VQ · ZR62"),
        new MockDevice("dev01",  "DEV-01 · Win 11",  "工位机", MockKind.Win,     0.88, 320, "#9AD0FF", "D1", "Windows 11", 41,
            Ip:"192.168.1.42", Bw:"940 Mbps", Fingerprint:"ZX8K · L72M · 9FQ3 · 7HD2 · M1P6 · QA8N · KZ9R · X3WF"),
    };

    public static readonly IReadOnlyList<MockHistory> History = new[]
    {
        new MockHistory("h6", "incoming", "孟茜", "14:18", MockHistoryKind.Image, Count: 2, Status: "done"),
        new MockHistory("h5", "outgoing", "孟茜", "14:10", MockHistoryKind.File, Name: "设计稿_v3_final.fig", Size: "14.2 MB", Ext: "fig", Status: "done"),
        new MockHistory("h4", "outgoing", "李莉", "14:09", MockHistoryKind.Text, Content: "改完了，整理一下发你 ", Status: "done"),
        new MockHistory("h3", "outgoing", "嘉伟", "14:08", MockHistoryKind.File, Name: "iOS-mocks-final.zip", Size: "48.6 MB", Ext: "zip", Progress: 67, Status: "transferring"),
        new MockHistory("h2", "incoming", "坤",   "13:58", MockHistoryKind.File, Name: "IMG_4821~38.heic",    Size: "128 MB",  Ext: "heic", Progress: 12, Status: "transferring"),
        new MockHistory("h1", "outgoing", "李莉", "13:42", MockHistoryKind.File, Name: "demo-video.mp4",      Size: "512 MB",  Ext: "mp4", Status: "queued"),
    };

    public static readonly MockPendingPairing PendingPairing = new(
        "pp-1", "李莉", "Lily's MacBook",
        "ZX8K · L72M · 9FQ3 · 7HD2 · M1P6 · QA8N · KZ9R · X3WF",
        "8s ago");

    public static readonly MockPendingOffer PendingOffer = new(
        "po-1", "嘉伟", "Jiawei · iPad",
        "规划文档_v0.3.pages", "3.4 MB",
        "改完了帮我看下第二章，特别是 §2.3 那段",
        "just now");

    public static readonly IReadOnlyList<MockClipboard> Clipboard = new[]
    {
        new MockClipboard("cb1", "嘉伟", MockClipboardKind.Link, "https://internal.acme.io/specs/auth-v3", "8s"),
        new MockClipboard("cb2", "孟茜", MockClipboardKind.Text, "1. 新流程要支持端到端\n2. 雷达扫描频率调到 2s\n3. iPad 端做横屏适配", "12m"),
        new MockClipboard("cb3", "李莉", MockClipboardKind.Code, "docker run --rm -v $PWD:/app meshdrop/build:latest", "34m", Lang:"sh"),
        new MockClipboard("cb4", "坤",   MockClipboardKind.Text, "会议室 B 已订到 16:00–17:30", "1h"),
        new MockClipboard("cb5", "我",   MockClipboardKind.Link, "figma://file/Q8xK2/MeshDrop?node-id=42:108", "2h"),
    };

    public static readonly IReadOnlyList<MockTransfer> Transfers = new[]
    {
        new MockTransfer("设计稿_v3_final.fig",  "14.2 MB",        "fig",  "我",   "孟茜",   100, MockTransferState.Done,                   Eta:"00:08"),
        new MockTransfer("iOS-mocks-final.zip",   "48.6 MB",        "zip",  "我",   "孟茜",   67,  MockTransferState.Sending,   "8.4 MB/s",   Eta:"00:02"),
        new MockTransfer("spec_PRD_2026Q1.pdf",   "2.1 MB",         "pdf",  "我",   "嘉伟",   34,  MockTransferState.Sending,   "3.1 MB/s",   Eta:"00:01"),
        new MockTransfer("IMG_4821~IMG_4838.heic","128 MB · 18 张", "heic", "坤",   "我",     12,  MockTransferState.Receiving, "11.7 MB/s",  Eta:"00:09"),
        new MockTransfer("release-notes.md",      "4.8 KB",         "md",   "我",   "DEV-01", 100, MockTransferState.Done,                   Eta:"00:01"),
        new MockTransfer("demo-video.mp4",        "512 MB",         "mp4",  "我",   "李莉",   0,   MockTransferState.Queued),
    };

    public static readonly int[] UploadBars   = { 3, 5, 8, 7, 9, 6, 11, 12, 14, 11, 10, 11, 12, 11 };
    public static readonly int[] DownloadBars = { 8, 9, 7, 6, 5, 7, 10, 12, 11, 12, 11, 12, 11, 12 };
    public static readonly int[] SessionBars  = { 2, 3, 5, 4, 6, 8, 7, 9, 10, 12, 11, 12, 11, 12, 14 };

    /// <summary>聊天页 mock 消息流（与李莉）。</summary>
    public static readonly IReadOnlyList<MockMessage> ChatWithLily = new[]
    {
        new MockMessage("m1", "in",  "text",  Text:"周末雷达扫描的卡顿排查得怎样？",                       Time:"13:32"),
        new MockMessage("m2", "out", "text",  Text:"找到了，mDNS QueryService 每 800ms 触发一次，改成 2.4s 之后稳定。", Time:"13:33"),
        new MockMessage("m3", "in",  "text",  Text:"赞 那这块就 freeze 了？",                                 Time:"13:33"),
        new MockMessage("m4", "out", "text",  Text:"是的。我把改后的对比图发你",                              Time:"13:34"),
        new MockMessage("m5", "out", "file",  FileName:"radar-diff.png",   FileSize:"384 KB",  FileExt:"png", Time:"13:35"),
        new MockMessage("m6", "in",  "text",  Text:"看了，比之前清爽多了。要不要顺便把 sweep 周期调成 4.5s？", Time:"13:38"),
        new MockMessage("m7", "out", "text",  Text:"已经是 4.5s 了 ",                                         Time:"13:39"),
        new MockMessage("m8", "out", "file",  FileName:"设计稿_v3_final.fig", FileSize:"14.2 MB", FileExt:"fig", Time:"14:10"),
    };

    public static readonly IReadOnlyList<MockTrust> Trusted = new[]
    {
        new MockTrust("李莉", "Lily's MacBook",   "macOS 15.3",   "ZX8K · L72M · 9FQ3 · 7HD2", "2026-04-21", "刚刚", true),
        new MockTrust("嘉伟", "Jiawei · iPad",    "iPadOS 18.2",  "BFM8 · 7KQ2 · X3LN · 9PRD", "2026-04-18", "14:08", true),
        new MockTrust("孟茜", "Meng Xi · iPhone", "iOS 18.2",     "H7LZ · 2K9F · QM48 · X1DR", "2026-03-09", "14:10", true),
        new MockTrust("坤",   "Kun · Pixel 8",    "Android 15",   "V2KN · 4HD8 · QX91 · LM3F", "2026-02-26", "13:58", true),
    };
}

/// <summary>给 XAML ItemsControl 用的可观察集合工厂（仅 designer preview 用）。</summary>
public static class MockObs
{
    public static ObservableCollection<MockDevice> Devices() => new(MockData.Devices);
    public static ObservableCollection<MockHistory> History() => new(MockData.History);
    public static ObservableCollection<MockClipboard> Clipboard() => new(MockData.Clipboard);
    public static ObservableCollection<MockTransfer> Transfers() => new(MockData.Transfers);
    public static ObservableCollection<MockMessage> ChatWithLily() => new(MockData.ChatWithLily);
    public static ObservableCollection<MockTrust> Trusted() => new(MockData.Trusted);
}
#endif
